topi/python/topi/cuda/dense.py

```diff
@@ -85,32 +89,136 @@ def schedule_dense(cfg, outs):
     """
     # pylint: disable=unused-argument
     target = tvm.target.current_target()
+
+    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
     if target.target_name == "cuda" and "cublas" in target.libs:
+        A, B = outs[0].op.input_tensors
+        b, i = get_const_tuple(A.shape)
+        o, _ = get_const_tuple(B.shape)
+        cfg.add_flop(2 * i * b * o)
         return generic.schedule_extern(outs)
 
-    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
     s = tvm.create_schedule([x.op for x in outs])
-    def _schedule(Dense):
-        num_thread = 64
-        k = Dense.op.reduce_axis[0]
-        ko, kf = s[Dense].split(k, factor=num_thread)
-        DenseF = s.rfactor(Dense, kf)
-
-        if Dense.op in s.outputs:
-            Out = Dense
+
+    def _schedule(C):
+        A, B = C.op.input_tensors
+        batch, in_dim = get_const_tuple(A.shape)
+        out_dim, _ = get_const_tuple(B.shape)
+        if batch < 64:
```

Comment:
Just curious, how would dynamic batching change this in the future? Would the batch dimension be fixed before entering here or this line will need to change? @comaniac @icemelon9

Reply:
My understanding is that the current plan for dynamic batching will determine the batch size before reaching to this point. Otherwise we will need to change all TOPI schedules which is too tedious.